specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2019-05-06-preview/searchindex.json

```diff
@@ -857,6 +902,27 @@
       "additionalProperties": true,
       "description": "A single bucket of a facet query result. Reports the number of documents with a field value falling within a particular range or having a particular value or interval."
     },
+    "SearchFeatures": {
+      "properties": {
+        "uniqueTokenMatches": {
+          "type": "number",
+          "readOnly": true,
+          "format": "double",
+          "description": "The number of unique tokens from the search query that matched this field."
```

Comment:
is this always a whole number? what does it being a decimal type mean?

Reply:
As of now, we only have those 2 features (unique tokens and similarity score). While we expect "unique tokens" to be whole numbers, almost all features we intend on won't be whole numbers. We also expect most customers to simply flatten out those structures into vectors of floats to be used for machine learning use cases.

Reply:
Gotcha. It's all `number` to JS fortunately for me. 😄

Reply:
If customers want to use this value in a vector of floats for ML training or prediction, that's super easy for them to do regardless of how we return the property.  My bigger concern is that every person using this API is going to wonder "Why is that a double?  Am I missing something here?"  That was my first thought too, anyway.

Reply:
Sorry for the late reply. That's good feedback, I can see people asking themselves the question. Our vision of the feature response was really just a bag of float describing the query-to-document relationship. The names are labels to describe how the values were calculated, but we never intended for those values to be used as discrete values. As we evolve it, there's even a chance that we start to apply normalization factors on those values to make them more useful. I don't necessarily mind converting some of those values to integer in the swagger if you think it would otherwise make their usage too confusing, but also see value in keeping them the way they are on the server-side.